File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-8.cpp
class Solution {
public:
    int characterReplacement(std::string s, int k) {
        int l = 0, res = 0, mf = 1;
        unordered_map<char,int> m;
        m[s[l]]++;
        for (int r =1; r<s.size(); r++) {
            m[s[r]]++;
            mf = max(mf, m[s[r]]);
            while(r-l+1-k>mf) {
                m[s[l]]--;
                l++;
            }
            res = max(res, r-l+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-8.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        l, r, m = 0, len(height)-1, 0
        lm, rm = height[l], height[r]
        while l < r:
            if lm < rm:
                l+=1
                lm = max(lm,height[l])
                m+=lm-height[l]
                
            else: 
                r-=1
                rm = max(rm,height[r])
                m+=rm-height[r]
                
        return m